### PR TITLE
feat(terraform): add CKV NCP rules about Load Balancer Listener Using HTTPS

### PR DESCRIPTION
**Pull request by pj991207**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Add terraform scan rules about provider Naver Cloud Platform.
CKV_NCP_24 is associated with Load Balancer Listener Using HTTPS

### Description
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/lb_listener

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/ncp/LBListenerUsingHTTPS.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class LBListenerUsingHTTPS(BaseResourceCheck):
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/ncp/LBListenerUsingHTTPS.py:7]: @pj991207 This should be a `BaseResourceValueCheck` inheritance IMO

Comment (pj991207) [reply checkov/terraform/checks/resource/ncp/LBListenerUsingHTTPS.py]: Thanks I reflect your advice

**Review by gruebel — APPROVED** [review]
🍺

File: checkov/terraform/checks/resource/ncp/LBListenerUsingHTTPS.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        id = "CKV_NCP_24"
+        supported_resources = ("ncloud_lb_listener",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def get_inspected_key(self):
+        return 'protocol'
+
+    def get_expected_values(self):
+        return ['HTTPS']
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/ncp/LBListenerUsingHTTPS.py:20]:
```suggestion
    def get_expected_value(self):
        return 'HTTPS'
```

**Review by nimrodkor — APPROVED** [review]
1 final suggestion @pj991207

**Pull request merged**